runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -406,6 +406,7 @@ ClientSessionData::getOrCacheVMInfo(JITServer::ServerStream *stream)
       auto recv = stream->read<VMInfo, std::vector<CacheDescriptor> >();
       _vmInfo = new (PERSISTENT_NEW) VMInfo(std::get<0>(recv));
       _vmInfo->_j9SharedClassCacheDescriptorList = reconstructJ9SharedClassCacheDescriptorList(std::get<1>(recv));
+      runtimeHelpers.setAddress(TR_prepareForOSR, _vmInfo->_prepareForOSRHelper);
```

Comment:
Is this `runtimeHelpers` global? If so, it's not going to work.

Reply:
Yeah, it's global. I don't see a good way of accessing this helper from OMR though, it's the only reason I set address here, instead of at the callsite.

Reply:
I guess one way would be to add a method to the front-end to specifically get `prepareForOSR` helper, I just don't find this to be a particularly nice solution